changelog/12472.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fixed a crash when returning category ``"error"`` or ``"failed"`` with a custom test status from :hook:`pytest_report_teststatus` hook -- :user:`pbrezina`.
```

Comment:
FTR here's the change note preview: https://pytest--12472.org.readthedocs.build/en/12472/changelog.html#bug-fixes